src/poc/miner_poc_statem.erl

```diff
@@ -159,22 +159,34 @@ requesting(info, Msg, #data{blockchain = Chain} = Data) when Chain =:= undefined
         NewChain ->
             {keep_state, Data#data{blockchain=NewChain}, [{next_event, info, Msg}]}
     end;
-requesting(info, {blockchain_event, {add_block, BlockHash, false, Ledger}}, #data{address=Address}=Data) ->
-    case allow_request(BlockHash, Data) of
-        false ->
-            lager:debug("request not allowed @ ~p", [BlockHash]),
-            {keep_state, Data};
-        true ->
-            {Txn, Keys, Secret} = create_request(Address, BlockHash, Ledger),
-            #{public := PubKey} = Keys,
-            <<ID:10/binary, _/binary>> = libp2p_crypto:pubkey_to_bin(PubKey),
-            lager:md([{poc_id, blockchain_utils:bin_to_hex(ID)}]),
-            lager:info("request allowed @ ~p", [BlockHash]),
-            ok = blockchain_worker:submit_txn(Txn),
-            lager:info("submitted poc request ~p", [Txn]),
-            {next_state, mining, save_data(Data#data{state=mining, secret=Secret,
-                                                      onion_keys=Keys, responses=#{},
-                                                      poc_hash=BlockHash})}
+requesting(info, {blockchain_event, {add_block, BlockHash, Sync, Ledger}} = Msg,
+           #data{address=Address}=Data) ->
+    Now = erlang:system_time(seconds),
+    case blockchain:get_block(BlockHash, Data#data.blockchain) of
+        {ok, Block} ->
+            case Sync andalso (Now - blockchain_block:time(Block) > 3600) of
```

Comment:
So if this is not a sync block OR it's a sync block newer than 1 hour, we try to make progress?

Reply:
yes.  the code comes directly from the consensus manager

Reply:
If we need to duplicate it again, lets make a utility function.